File: 240219/배열의 평균/ave-of-array.py
grid = [list(map(int, input().split()))
for _ in range(2)]

# 가로 평균
for i in range(2):
  sum_val = 0
  for j in range(4):
    sum_val += grid[i][j]
  print(sum_val/4, end = ' ')
print()

# 새로 평균
for i in range(4):
  sum_val = 0
  for j in range(2):
    sum_val += grid[j][i]
  print(sum_val/2, end= ' ')
print()

# 전체 평균
total = 0
for i in range(2):
  for j in range(4):
    total += grid[i][j]
print(f"{total/8:.1f}")

File: 240409/배열의 평균/ave-of-array.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split(`\n`)

let grid = []

for (let i = 0; i < 2; i++ ) {
    grid.push(input[i].split(' ').map(Number))
}

let answerFirst = []
for (let i = 0; i < 2; i ++ ) {
    let sum = 0
    for (let j = 0; j < 4; j ++ ) {
        sum += grid[i][j]
    }
    answerFirst.push((sum/4).toFixed(1))
}
console.log(answerFirst.join(' '))

let answerSecond = []
for (let i = 0; i < 4; i ++ ) {
    let sum = 0
    for (let j = 0; j < 2; j ++ ) {
        sum += grid[j][i]
    }
    answerSecond.push((sum/2).toFixed(1))
}

console.log(answerSecond.join(' '))

let answerThird = 0

for (let i=0; i<2; i++ ) {
    for (let j=0; j<4; j++) {
        answerThird += grid[i][j]
    }
}

console.log((answerThird/8).toFixed(1))